flink-cdc-connect/flink-cdc-source-connectors/flink-cdc-base/src/main/java/org/apache/flink/cdc/connectors/base/source/meta/split/SourceSplitSerializer.java

```diff
@@ -199,7 +199,7 @@ public static void writeTableSchemas(
             boolean useCatalogBeforeSchema =
                     SerializerUtils.shouldUseCatalogBeforeSchema(entry.getKey());
             out.writeBoolean(useCatalogBeforeSchema);
-            out.writeUTF(entry.getKey().toString());
+            out.writeUTF(entry.getKey().toDoubleQuotedString());
```

Comment:
@edmond-kk @loserwang1024 Is this a compatible change?

Reply:
If without this change, NullPointerException will be thrown. We have done it in  out.writeUTF(snapshotSplit.getTableId().toDoubleQuotedString()) but not in writeTableSchemas.
<img width="793" alt="image" src="https://github.com/user-attachments/assets/996fa365-e10c-4c90-aab8-7d9e667751fc">